rust_snuba/src/metrics/statsd.rs

```diff
@@ -1,15 +1,28 @@
-use cadence::prelude::*;
-use cadence::{MetricBuilder, MetricError, StatsdClient};
-use rust_arroyo::utils::metrics::Metrics as ArroyoMetrics;
+use std::collections::HashMap;
+
+use rust_arroyo::metrics::{Metric, MetricSink, Recorder, StatsdRecorder};
 use statsdproxy::cadence::StatsdProxyMetricSink;
 use statsdproxy::config::AggregateMetricsConfig;
 use statsdproxy::middleware::aggregate::AggregateMetrics;
 use statsdproxy::middleware::Upstream;
-use std::collections::HashMap;
 
 #[derive(Debug)]
 pub struct StatsDBackend {
-    client: StatsdClient,
+    recorder: StatsdRecorder<Wrapper>,
+}
+
+impl Recorder for StatsDBackend {
+    fn record_metric(&self, metric: Metric<'_>) {
+        self.recorder.record_metric(metric)
+    }
+}
+
+struct Wrapper(Box<dyn cadence::MetricSink + Send + Sync + 'static>);
+
+impl MetricSink for Wrapper {
```

Comment:
Couldn't this be implemented directly on the `Box<dyn …>`?

Reply:
Not here at least, but rather where `MetricSink` is implemented due to orphan rules.

Reply:
Oh, sorry, I missed that this wasn't happening in `rust_arroyo`. Should the `cadence` compatibility stuff perhaps be moved there?

Reply:
Sounds reasonable